### PR TITLE
Add `Multisig` and `MultisigWatchOnly` identities

### LINKED ISSUE #80: Add Multisignature BIP48 identity

### LINKED ISSUE #81: Use nigiri-github-action in CI

### PR DESCRIPTION
**Pull request by louisinger**
This PR implements two `IdentityInterface` using to handle multi-signature cases.

**Common**

* add a static `Random` constructor in `Mnemonic` identity.
* abstract the base derivation path in `MasterPublicKey` (via an optional argument in the constructor).
* add optional base derivation path argument in `Mnemonic`.

**MultisigWatchOnly**

* and identity **unable to sign transactions**.
* it's the equivalent of `MasterPubKey` for `Mnemonic`.
* it lets to generate deterministic multi-sig addresses. 
* in order to create a `MultiSigWatchOnly` instance, u need to provide a `MultisigWatchOnlyOpts` i.e a list of xpubs and a number (the number of required signatures).
* generate the blinding private key using the XOR trick.
* tested in new file `multisigwatchonly.identity.test.ts`.

**Multisig**

* extends `MultisigWatchOnly`.
* able to sign transactions.
* cache script to derivation path in order to find inputs to sign in `signPset`.
* in order to create a `Multisig` you need to provide (1) the signer mnemonic and (2) the list of the **others** cosigners. 
* tested in new file `multisig.identity.test.ts`.

### Usage

This:
```typescript
const watchOnly = new MultisigWatchOnly({
  chain: 'regtest',
  type: IdentityType.MultisigWatchOnly,
  opts: {
    cosigners: ['xpub01...', 'xpub02...', 'xpub03...'],
    requiredSignatures: 2, // will create a 2-of-3
}});
```
...is the watchOnly of:
```typescript
const multisig = new Multisig({
  chain: 'regtest',
  type: IdentityType.Multisig,
  opts: {
    signer: { mnemonic: "this is the mnemonic words of xpub01" },
    cosigners:  ['xpub02...', 'xpub03...'],
    requiredSignatures: 2, // will create a 2-of-3
}});
```

it closes #80 
it closes #81 

@tiero please review

### CODE REVIEW

**Review by tiero — COMMENTED** [review]
re: cosigners 

I would prefer to have an array of typed objects, so we the order how we pass the xpubs does not matter and also we can have personal multisig where we all own the keys (ie. Specter like).

Description is general useful to tag a specific cosigner such as physical location tips or a specific person/entity owning that.

```
export interface Cosigner {
  xpub: string;
  isMine: boolean;
  signer?: Signer | Mnemonic ;
  description?: string;
}
```

EDIT: Signer can be for example a class that handle the calls with an external hardware wallet for example. For now we can stick to Mnemonic/string only

**Comment by Janaka-Steph** [comment]
Can check what has been done on Equi App btw https://github.com/equitas-foundation/equi-gate-ui/blob/master/src/features/transactions/PendingTransactionItem.tsx#L56

**Comment by louisinger** [comment]
UPDATE: type the arguments of Multisig Identity

```typescript
export interface SignerMultisig {
  mnemonic: string;
  baseDerivationPath?: string;
}

export type XPub = string;

export type CosignerMultisig = XPub | SignerMultisig; /
```

the `cosigners` field is an Array of `CosignerMultisig`.
the `signer` field is an Array of `SignerMultisig`.

File: src/explorer/transaction.ts [hunk 1]
```diff
@@ -47,7 +53,15 @@ export async function* fetchAndUnblindTxsGenerator(
...
-      errors.push(err);
+      if (err instanceof Error) {
+        errors.push(err);
+      }
+
+      if (typeof err === 'string') {
+        errors.push(new Error(err));
+      }
+
+      errors.push(new Error('unknow error'));
```
Comment (tiero) [inline src/explorer/transaction.ts:64]:
```suggestion
      errors.push(new Error('unknown error'));
```

File: src/types.ts [hunk 2]
```diff
@@ -80,6 +94,20 @@ export interface TxInterface {
...
+  vin: InputInterface[];
+  vout: (BlindedOutputInterface | UnblindedOutputInterface)[];
 }
+
+export type MultisigPayment = AddressInterface & {
+  redeemScript: string;
+  witnessScript: string;
+};
+
+export interface SignerMultisig {
```
Comment (tiero) [inline src/types.ts:106]: I would call it 
```suggestion
export interface HDSignerMultisig {
```

**Pull request merged**